pkg/operator/hive/hive_controller.go

```diff
@@ -522,6 +544,21 @@ func (r *ReconcileHiveConfig) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	// If we get here, we've successfully scrubbed all *our* resources out of previous target
+	// namespaces. Ideally, we would delete the namespace. Unfortunately, there's no good way to
+	// tell whether a) we were the one to create it, or b) it's empty. So settle for unlabeling it,
+	// accepting the fact that we might be leaking namespaces.
```

Comment:
BTW

> there could only ever be one "previous" namespace

It would take serious effort to contrive a scenario where this was not the case; but for example, I change from A=>B and A can't get fully cleaned up for some reason before I change from B=>C; now I have two stale namespaces.

> we can store in hiveConfig.status.currentNamespace

I love this idea... but the current code also accounts for "changing" the target namespace by deleting the existing HiveConfig and creating a whole new one.

Reply:
Yeah, that "delete HiveConfig" scenario would have to be protected by a finalizer so we don't lose track of the status. Just trying to keep things clean rather than appeal to these diabolical scenarios :)

Reply:
To be clear: are you asking for something to be changed in this PR, or are we still talking hypothetical future (maybe v2) refactoring?

Reply:
Yeah, this is v2 stuff. We'll make do with what we have now